csi.proto

```diff
@@ -1008,6 +1053,303 @@ message GetCapacityResponse {
   google.protobuf.Int64Value minimum_volume_size = 3
     [(alpha_field) = true];
 }
+message CreateVolumeGroupRequest {
+  option (alpha_message) = true;
+
+  // suggested name for volume group (required for idempotency)
+  // This field is REQUIRED.
+  string name = 1;
+
+  // params passed from VolumeGroupClass
+  // This field is OPTIONAL.
+  map<string, string> parameters = 2;
+
+  // Secrets required by plugin to complete volume group creation
+  // request.
+  // This field is OPTIONAL. Refer to the `Secrets Requirements`
+  // section on how to use this field.
+  map<string, string> secrets = 3 [(csi_secret) = true];
+
+  // Phase 2
+  // If specified, a volume group will be created from the source group
+  // snapshot.
+  // This field is OPTIONAL.
+  // VolumeGroupSnapshot source_volume_group_snapshot = 4;
+
+  // Phase 2
+  // If specified, a volume group will be created from a list of
+  // existing volumes.
+  // This field is OPTIONAL.
+  // repeated string volume_id = 5;
+}
+
+message CreateVolumeGroupResponse {
+  option (alpha_message) = true;
+
+  // Contains all attributes of the newly created volume group.
+  // This field is REQUIRED.
+  VolumeGroup volume_group = 1;
+}
+
+message VolumeGroup {
+  option (alpha_message) = true;
+
+  // The identifier for this volume group, generated by the plugin.
+  // This field is REQUIRED.
+  string volume_group_id = 1;
+
+  // Opaque static properties of the volume group.
+  // This field is OPTIONAL.
+  map<string, string> volume_group_context = 2;
+
+  // Underlying volumes in this group. The same definition in CSI
+  // Volume.
+  // This field is REQUIRED.
+  // To support the creation of an empty group, this list can be empty.
+  // However, this field is not empty in the following cases:
+  // - Response from ListVolumeGroups or GetVolumeGroup if the
+  //   VolumeGroup is not empty.
+  // - Response from ModifyVolumeGroup if the VolumeGroup is not
+  //   empty after modification.
+  // - Phase 2: Create a new volume group from a source group snapshot.
+  // - Phase 2: Create a new volume group and add a list of existing
+  //   volumes to the group.
+  repeated .csi.v1.Volume volumes = 3;
+}
+message DeleteVolumeGroupRequest {
+  option (alpha_message) = true;
+
+  // The ID of the volume group to be deprovisioned.
```

Comment:
good to mention `volume group to be deleted` to be consistent with next field explanation.